webapp/src/ts/services/target-aggregates.service.ts

```diff
@@ -304,28 +325,26 @@ export class TargetAggregatesService {
     return aggregates.find(aggregate => aggregate.id === targetId);
   }
 
-  getCurrentTargetDoc(contact?) {
-    return this.ngZone.runOutsideAngular(() => this._getCurrentTargetDoc(contact));
+  getTargetDocs(contact, userFacilityId, userContactId) {
+    return this.ngZone.runOutsideAngular(() => this._getTargetDocs(contact, userFacilityId, userContactId));
   }
 
-  private _getCurrentTargetDoc(contact?) {
-    if (!contact) {
-      return Promise.resolve();
+  private async _getTargetDocs(contact, userFacilityId, userContactId) {
+    const contactUuid = contact?._id;
+    if (!contactUuid) {
+      return;
     }
 
-    const contactUuid = _isString(contact) ? contact : contact._id;
-
-    if (!contactUuid) {
-      return Promise.resolve();
+    const settings = await this.settingsService.get();
```

Comment:
Minor: Move this down to just before it's needed and avoid unnecessary work if the `return` fires.

Reply:
Done. At some point I had passed it to contact types service to check the contact type, but changed that code afterwards.